src/gt4py/next/program_processors/runners/dace_iterator/utility.py

```diff
@@ -181,6 +181,24 @@ def unique_var_name():
     return unique_name("_var")
 
 
+def map_field_dimensions_to_sdfg_symbols(
```

Comment:
For some reason I do not like the name of this function, since it does not express what the function is actually doing.
My suggestion would be something  `_create_shape_stride()`.
- Since it is internal to the package we would prefix it with `_`.
- I do not like the word mapping in this context, since for me a mapping, as it is used in the context here (to my knowledge; I know there is `map`), establishes a connection between two "things". However, here if needed we create the symbols, so I think we should include [creat](https://linux.die.net/man/2/creat) or make in the function name.

If you want to keep the name, then you can ignore the stuff above.


However, the function does something highly non trivial, so we should add a meaningful docstring or at least some comments.
I do not believe in the ["the code is the ultimate documentation" mantra](https://web.mit.edu/~simsong/www/ugh.pdf). <!-- Although I love UNIX. -->

Reply:
I agree, function name changed to `_make_array_shape_and_strides`, doc string added and function moved to file scope.